app/controllers/books_controller.rb

```diff
@@ -1,31 +1,44 @@
 class BooksController < ApplicationController
-  
-  require 'net/http'
-  require 'json'
+  require "net/http"
+  require "json"
 
   def search
     @query = params[:query]
-    @url = "http://openlibrary.org/search.json?q=#{@query}&limit=10"
+    @url = "http://openlibrary.org/search.json?q=#{@query}&limit=12"
     @uri = URI(@url)
     @response = Net::HTTP.get(@uri)
     @output = JSON.parse(@response)
-    @books = @output['docs']
-    # @pretty = JSON.pretty_generate(@books)
-    # render plain: @pretty
+    @books = @output["docs"]
   end
 
-  def show 
+  def show
     @id = params[:id]
-    @isbn = params[:isbn]
-    @author_url = "https://openlibrary.org/api/books?bibkeys=ISBN:#{@isbn}&jscmd=details&format=json" 
+
+    @review = Review.new
+
     @url = "https://openlibrary.org/works/#{@id}.json"
     @uri = URI(@url)
-    @author_uri = URI(@author_url)
-    @author_response = Net::HTTP.get(@author_uri) 
     @response = Net::HTTP.get(@uri)
     @book = JSON.parse(@response)
-    @author = JSON.parse(@author_response)["ISBN:#{@isbn}"]['details']['authors'][0]['name']
-  end
 
+    @isbn = params[:isbn]
+    @author_url = "https://openlibrary.org/api/books?bibkeys=ISBN:#{@isbn}&jscmd=details&format=json"
+    @author_uri = URI(@author_url)
+    @author_response = Net::HTTP.get(@author_uri)
 
+    @author_json = JSON.parse(@author_response)["ISBN:#{@isbn}"]["details"]["authors"]
+
+    if (@author_json == nil)
+      @suggestions = []
+      @author = []
+    else
+      @author = @author_json[0]["name"]
+      @suggestion_url = "http://openlibrary.org/search.json?author=#{@author}&limit=3"
+      @suggestion_uri = URI(@suggestion_url)
+      @suggestion_response = Net::HTTP.get(@suggestion_uri)
+      @suggestions = JSON.parse(@suggestion_response)["docs"]
+    end
+    @preview = JSON.parse(@author_response)["ISBN:#{@isbn}"]["preview_url"]
+    @reviews = Review.where(book_id: "#{@id}")
```

Comment:
I don't think we need interpolation here, you can directly use variable. Can't we do this ?
```suggestion
    @reviews = Review.where(book_id: @id)
```

Reply:
You're right we don't need that, so I will fix that, also I will use local variable instead of instance variable.